clouddriver/clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/docker/HelmOciDockerArtifactAccount.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2025 Harness, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.artifacts.docker;
+
+import com.google.common.base.Strings;
+import com.netflix.spinnaker.clouddriver.artifacts.config.ArtifactAccount;
+import com.netflix.spinnaker.kork.annotations.NonnullByDefault;
+import com.netflix.spinnaker.kork.client.ServiceClientProvider;
+import com.netflix.spinnaker.kork.docker.service.DefaultDockerOkClientProvider;
+import com.netflix.spinnaker.kork.docker.service.DockerRegistryClient;
+import java.io.File;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.ParametersAreNullableByDefault;
+import lombok.Builder;
+import lombok.Value;
+import org.springframework.boot.context.properties.ConstructorBinding;
+
+@NonnullByDefault
+@Value
+public class HelmOciDockerArtifactAccount implements ArtifactAccount {
+  private final String name;
+
+  private final String username;
+  private final String password;
+  private final File passwordFile;
+  private final String passwordCommand;
+  private final File dockerconfigFile;
+  private final String email;
+  private final String address;
+  private final boolean insecureRegistry;
+  private final List<String> helmOciRepositories;
+  private final long clientTimeoutMillis;
+  private final int paginateSize;
+
+  private final ServiceClientProvider serviceClientProvider;
+
+  private final String type = HelmOciDockerArtifactCredentials.CREDENTIALS_TYPE;
+  private final String provider = HelmOciDockerArtifactCredentials.TYPE;
+
+  @Builder
+  @ConstructorBinding
+  @ParametersAreNullableByDefault
+  HelmOciDockerArtifactAccount(
```

Comment:
A LOT of these are duplciated from the base Docker credentials... can this just extend it, use SuperBuilder & allow removing a lot of duplicated properties?

Reply:
pushed a TODO in all three classes